### DISCUSSION TITLE
Appending Element to dom at initialized stage

### DISCUSSION THREAD

**Discussion by timiosaze** [alpinejs/alpine#3172] [4. General]
```
init() {
            console.log("he is  there"), // works
            document.getElementsByTagName("span").append("<h2>yellow</h2>") //does not work
            $("span").append("<h2>yellow</h2>") //does not work
        },
```
I want to know why they dont work

**Comment by ekwoka** [comment]
What is your actual goal?

One issue is that `getElementsByTagName` returns an html element collection. Which does not have an append method.

Which the console errors would tell you.

So please actually read the console if something is not working.

But this is likely not the way you should be doing this anyway.

**Reply by timiosaze** [reply]
please which way should i be doing it @ekwoka

**Reply by ekwoka** [reply]
Just put the element in the markup....
